pallets/asset/src/lib.rs

```diff
@@ -2066,160 +2110,121 @@ impl<T: Config> Module<T> {
-        // No check since the total balance is always <= the total supply. The
-        // total supply is already checked above.
-        let updated_to_balance = current_to_balance + value;
-        // No check since the portfolio balance is always <= the total
-        // supply. The total supply is already checked above.
-        let updated_to_def_balance =
-            Portfolio::<T>::portfolio_asset_balances(issuer_portfolio, ticker) + value;
-
-        // In transaction because we don't want fee to be charged if advancing fails.
-        with_transaction(|| {
-            // Charge the fee.
-            if let Some(op) = protocol_fee_data {
-                T::ProtocolFee::charge_fee(op)?;
-            }
+        let current_issuer_balance = BalanceOf::get(&ticker, &issuer_portfolio.did);
+        <Checkpoint<T>>::advance_update_balances(
+            &ticker,
+            &[(issuer_portfolio.did, current_issuer_balance)],
+        )?;
 
-            <Checkpoint<T>>::advance_update_balances(
-                ticker,
-                &[(issuer_portfolio.did, current_to_balance)],
-            )
-        })?;
+        let new_issuer_balance = current_issuer_balance + amount_to_issue;
+        BalanceOf::insert(ticker, issuer_portfolio.did, new_issuer_balance);
 
-        // Increase total supply.
-        token.total_supply = updated_total_supply;
-        BalanceOf::insert(ticker, issuer_portfolio.did, updated_to_balance);
-        Portfolio::<T>::set_portfolio_balance(issuer_portfolio, ticker, updated_to_def_balance);
-        Tokens::insert(ticker, token);
+        security_token.total_supply += amount_to_issue;
+        Tokens::insert(ticker, security_token);
+
+        let new_issuer_portfolio_balance =
+            Portfolio::<T>::portfolio_asset_balances(issuer_portfolio, ticker) + amount_to_issue;
+        Portfolio::<T>::set_portfolio_balance(
+            issuer_portfolio,
+            &ticker,
+            new_issuer_portfolio_balance,
+        );
 
         Statistics::<T>::update_asset_stats(
             &ticker,
             None,
             Some(&issuer_portfolio.did),
             None,
-            Some(updated_to_balance),
-            value,
+            Some(new_issuer_balance),
+            amount_to_issue,
             weight_meter,
         )?;
 
-        let round = Self::funding_round(ticker);
-        let ticker_round = (*ticker, round.clone());
-        // No check since the issued balance is always <= the total
-        // supply. The total supply is already checked above.
-        let issued_in_this_round = Self::issued_in_funding_round(&ticker_round) + value;
-        IssuedInFundingRound::insert(&ticker_round, issued_in_this_round);
+        let funding_round_name = FundingRound::get(&ticker);
+        IssuedInFundingRound::mutate((&ticker, &funding_round_name), |balance| {
+            *balance += amount_to_issue
+        });
```

Comment:
We should either check for overflow and throw an error, or use `saturing_add`.  This would only be an issue if `MAX_SUPPLY` was increase to be much closer to `u128::MAX`.